\begin{center}
Gabe Morris
\end{center}

In [1]:
# Notebook Preamble
import matplotlib.pyplot as plt
import sympy as sp

plt.style.use('maroon_ipynb.mplstyle')

def von_mises(sx_, sy_, sz_, txy_, tyz_, tzx_):
    return(1/sp.sqrt(2)*sp.sqrt((sx_ - sy_)**2 + (sy_ - sz_)**2 + (sz_ -sx_)**2 + 6*(txy_**2 + tyz_**2 + tzx_**2))).n()

\pagebreak
\tableofcontents
\pagebreak

\begin{center}
\begin{tabular}{c c c}
ME 4403 & Homework 5 & Gabe Morris \\
& & gnm54
\end{tabular}
\end{center}

# Problem 8-1
## Given
A power screw is 25 mm in diameter and has a thread pitch of 5 mm.

In [2]:
d = sp.S(25)
p = sp.S(5)

## Find
a. Find the thread depth, the thread width, the mean and root diameters, and the lead, provided square threads are used.
b. Repeat part (a) for Acme threads.

## Solution

\begin{center}
\includegraphics{images/fig1}
\end{center}

### Part A
**Thread Depth**

From (a) in the figure above, the thread depth is half the pitch.

In [3]:
# Thread depth
thread_depth = (p/2).n()
thread_depth  # mm

2.50000000000000

**Thread Width**

The thread width is the same as the thread depth.

In [4]:
thread_width = thread_depth
thread_width  # mm

2.50000000000000

**Mean and Root Diameters**

In [5]:
# Root diameter is the same as minor diameter
dr = d - 2*thread_depth
dr  # mm

20.0000000000000

In [6]:
# Mean diameter
dm = (d + dr)/2
dm  # mm

22.5000000000000

**Lead**

In [7]:
# The lead is the same as the pitch because it's single threaded
l = p
l  # mm

5

### Part B
The same procedure is done in Part B, but now we look at (b) in the above figure.

**Thread Depth and Thread Width**

In [8]:
# The thread depth and thread width is still the same
thread_depth  # mm

2.50000000000000

In [9]:
thread_width  # mm

2.50000000000000

Because the thread depth is the same, so will the minor and mean diameter. The lead also remains unchanged.

**Mean Diameter, Root Diameter, and Lead**

In [10]:
dm  # mm

22.5000000000000

In [11]:
dr  # mm

20.0000000000000

In [12]:
l  # mm

5

\pagebreak

\begin{center}
\begin{tabular}{ccc}
ME 4403 & Homework 5 & Gabe Morris \\
& & gnm54
\end{tabular}
\end{center}

# Problem 8-7
## Given

\begin{center}
\includegraphics{images/fig2}
\end{center}

For the C clamp shown, a force is applied at the end of the $\frac{3}{8}\,in$ diameter handle. The screw is a $\frac{3}{4}\,in-6$ Acme thread, and is 10 inches long overall, with a maximum of 8 inches possible in the clamping region. The handle and screw are both made from cold-drawn AISI 1006 steel. The coefficients of friction for the screw and the collar are 0.15. The collar, which in this case is the anvil striker’s swivel joint, has a friction diameter of 1 inch. It is desired that the handle will yield before the screw will fail.

In [13]:
# Cold Drawn Steel
Sy = sp.S(41)  # ksi

## Find
Check this by the following steps:

a. Determine the maximum force that can be applied to the end of the handle to reach the point of yielding of the handle.
b. Using the force from part (a), determine the clamping force.
c. Using the force from part (a), determine the factor of safety for yielding at the interface of the screw body and the base of the first engaged thread, assuming the first thread carries 38% of the total clamping force.
d. Determine the factor of safety for buckling of the screw.

## Solution
### Part A
The handle is a cantilever beam.

In [14]:
# Getting the maximum force
F = sp.Symbol('F')
M = (sp.S('3.5') - sp.S('0.375'))*F
d_handle = sp.Rational(3, 8)
I = sp.pi*d_handle**4/64
eq1 = sp.Eq((M*(d_handle/2)/I).nsimplify(), Sy)  # Force in kips
eq1

Eq(51200*F/(27*pi), 41)

In [15]:
F_ = (sp.solve(eq1)[0]*1000).n()
F_  # lbf

67.9246692875762

### Part B
For a $\frac{3}{4}\,in-6$ thread,

In [16]:
p = l = sp.Rational(1, 6).n()
d = sp.S('0.75')
dm = d - p/2
dm # inches

0.666666666666667

Equations 8-5 and 8-6 may be used to compute the force,

$$T_{R}=\frac{F d_{m}}{2}\left(\frac{l+\pi f d_{m} \sec \alpha}{\pi d_{m}-f l \sec \alpha}\right)$$
$$T_c=\frac{Ff_cd_c}{2}$$

In [17]:
fc = sp.S('0.15')
dc = 1
alpha = sp.rad(sp.S('14.5'))  # see Figure 8-7 (a)
sec_a = (1/sp.cos(alpha)).n()
T_ = sp.S('3.5')*F_
T = F*dm/2*((l + sp.pi*fc*dm*sec_a)/(sp.pi*dm - fc*l*sec_a)) + F*fc*dc/2
eq2 = sp.Eq(T, T_)
eq2

Eq(0.075*F + 0.333333333333333*F*(0.166666666666667 + 0.103290031219769*pi)/(-0.0258225078049421 + 0.666666666666667*pi), 237.736342506517)

In [18]:
F_clamp = sp.solve(eq2)[0]
F_clamp  # lbf

1542.27366292139

### Part C
Use Eq. 8-8, 8-11, 8-7, and 8-12.

In [19]:
# Finding the axial stress
dr = d - p
sig_y = ((-4*F_clamp/(sp.pi*dr**2))/1000).n()
sig_y  # ksi

-5.77082590952346

In [20]:
# Find the bending stress
sig_x = ((6*sp.S('0.38')*F_clamp/(sp.pi*dr*1*p))/1000).n()
sig_x  # ksi

11.5127976894993

In [21]:
tau_yz = ((16*T_/(sp.pi*dr**3))/1000).n()
tau_yz  # ksi

6.09979591836735

In [22]:
tau_zx = ((-4*sp.S('0.38')*T_/(sp.pi*dr**2*1*p))/1000).n()
tau_zx  # ksi

-2.02818214285714

Now the von mises stress may be calculated.

In [23]:
sig_prime = von_mises(sig_x, sig_y, 0, 0, tau_yz, tau_zx)
sig_prime  # ksi

18.8745435088120

In [24]:
# Get the factor of safety
ny = Sy/sig_prime
ny

2.17223796595972

### Part D
The critical force, $P_{cr}$, may be found from Eq. 4-48,

$$
\frac{P_{\mathrm{cr}}}{A}=S_{y}-\left(\frac{S_{y}}{2 \pi} \frac{l}{k}\right)^{2} \frac{1}{C E}
$$

$C=1.2$ from Table 4-2

In [25]:
C = sp.S('1.2')
E = sp.S(30_000)  # ksi (steel)
k = dr/4
A = sp.pi/4*dr**2
P_cr = (A*(Sy - (Sy*8/(2*sp.pi*k))**2*1/(C*E))).n()
P_cr

10.0061437991992

In [26]:
# Factor of safety
P_cr*1000/F_clamp

6.48791718341702

\pagebreak

\begin{center}
\begin{tabular}{ccc}
ME 4403 & Homework 5 & Gabe Morris \\
& & gnm54
\end{tabular}
\end{center}

# Problem 8-9
## Given
A 1.5 inch power screw has double square threads with a pitch of $\frac{1}{4}\,in$. The nut is to move at a velocity of 2 in/s and move a load of $F=2.2\,kips$. The frictional coefficients are 0.1 for the threads and 0.15 for the collar. The frictional diameter of the collar is 2.25 inches.

In [27]:
p = sp.S('0.25')  # inches
d = sp.S('1.5')  # inches
f, fc = sp.S('0.1'), sp.S('0.15')
V = 2  # in/s
F = sp.S('2200')  # lbf
dc = sp.S('2.25')  # inches

l = 2*p  # lead in inches
dm = d - p/2  # mean diameter in inches
dr = d - p  # minor diameter in inches

## Find
Find the power required to drive the power screw.

## Solution
Power is $P=T\omega$ and the torque for square threads when raising a load is (Eq. 8-1),

$$T_R=\frac{Fd_m}{2}\left(\frac{l+\pi fd_m}{\pi d_m-fl}\right)$$

The total torque also encompasses Eq. 8-6.

In [28]:
TR = (F*dm/2*(l + sp.pi*f*dm)/(sp.pi*dm - f*l)).n()
TC = F*fc*dc/2
T = TR + TC
T  # in*lbf

701.391797325860

The angular velocity may be acquired by using the lead because in 1 revolution, the nut has travelled the distance of the lead.

In [29]:
# getting rev/s
w = V/l
w

4.00000000000000

In [30]:
# Getting power
hp = 1/(550*12)  # 1 hp = 550*12 in*lbf/s
P = T*w*2*sp.pi*hp
P.n()  # hp

2.67089371850553

\pagebreak

\begin{center}
\begin{tabular}{ccc}
ME 4403 & Homework 5 & Gabe Morris \\
& & gnm54
\end{tabular}
\end{center}

# Problem 8-11
## Given
An M14 $\times$ 2 hex-head bolt with a nut is used to clamp together two 15 mm steel plates.

## Find
a. Determine a suitable length for the bolt, rounded up to the nearest 5 mm.
b. Determine the bolt stiffness.
c. Determine the stiffness of the members.

## Solution
### Part A
Bolt dimensions may be found from Table A-31.

In [31]:
l, H = sp.S(15), sp.S('12.8')
L = 2*l + H
L

42.8000000000000

The bolt length must be greater than $42.8\,mm$ to the nearest $5\,mm$. The final value is $L=45\,mm$.

### Part B
The fastener stiffness along with more relationships come from Table 8-7.

$$k_b=\frac{A_dA_tE}{A_dl_t+A_tl_d}$$

In [32]:
# Getting stiffness
d = sp.S(14)  # mm
E = sp.S('207')  # Pa
L = 45  # mm

Ad = sp.pi*d**2/4
At = sp.S(115)
Lt = 2*d + 6
ld = L - Lt
lt = 2*l - ld
kb = Ad*At*E/(Ad*lt + At*ld)
kb.n()  # MN/m

874.618152674828

### Part C

In [33]:
# Eq. 8-22
x = sp.S('0.5774')
km = (x*sp.pi*E*d)/(2*sp.log(5*(x*2*l + sp.S('0.5')*d)/(x*2*l + sp.S('2.5')*d)))
km.n()  # MN/m

3116.45077847628

\pagebreak

\begin{center}
\begin{tabular}{ccc}
ME 4403 & Homework 5 & Gabe Morris \\
& & gnm54
\end{tabular}
\end{center}

# Problem 8-19
## Given
A 30 mm thick AISI 1020 steel place is sandwiched between two 10 mm thick 2024-T3 aluminum plates and compressed with a bolt and nut with no washers. The bolt is an M10 $\times$ 1.5, property class 5.8.

## Find
a. Determine a suitable length for the bolt, rounded up to the nearest 5 mm.
b. Determine the bolt stiffness.
c. Determine the stiffness of the members.

## Solution
### Part A
Use a similar procedure from 8-11.

In [34]:
l, H = sp.S(50), sp.S('8.4')
L_min = l + H
L_min  # mm

58.4000000000000

Rounding up to the nearest 5 mm, the length of the bolt is $L=60\,mm$.

### Part B

In [35]:
# From Table 8-7
L = sp.S(60)
d = sp.S(10)
E = sp.S(207)

LT = 2*d + 6
ld = L - LT
lt = l - ld
Ad = sp.pi*d**2/4
At = 58
kb = Ad*At*E/(Ad*lt + At*ld)
kb.n()  # MN/m

292.057923214078

### Part C
Use Eq. 8-20 and then acquire the equivalent stiffness.

In [36]:
def get_k(E_, d_, t_, D_):
    return ((sp.S('0.5774')*sp.pi*E_*d_)/sp.log(((sp.S('1.155')*t_ + D_ - d_)*(D_ + d_))/((sp.S('1.155')*t_ + D_ + d_)*(D_ - d_)))).n()

# The upper and lower plates have the same stiffness
k1 = get_k(71, 10, 10, 15)
k1  # MN/m

1576.11182584228

In [37]:
# splitting the top half
D2 = 15 + 20*sp.tan(30*sp.pi/180)
k2 = get_k(207, 10, 15, D2)
k2  # MN/m

11434.8600387284

In [38]:
k_half = 1/(1/k1 + 1/k2)
k_half

1385.18615838124

In [39]:
k_total = (1/k_half + 1/k_half)**-1
k_total  # MN/m

692.593079190621